### ISSUE TITLE
Unexpected output_layers after save_new_activations

### ISSUE THREAD

**Issue by raktes** [johnmarktaylor91/torchlens#40]
After running save_new_activations on an existing ModelHistory, the output_layers list is unexpectedly extended with the label of the last layer plus '_raw'.

Example code demonstrating the issue:
```
import torchlens as tl
import torch
from torch import nn

class BasicModel(nn.Module):
    def forward(self, x):
        return x

x = torch.rand(2, 3)
basic_model = BasicModel()
basic_mh = tl.log_forward_pass(basic_model, x)
print(basic_mh.output_layers) # ['output_1']

basic_mh.save_new_activations(basic_model, x)
print(basic_mh.output_layers) # ['output_1', 'input_1_raw']

basic_mh.save_new_activations(basic_model, x)
print(basic_mh.output_layers) # ['output_1', 'input_1_raw', 'input_1_raw']


class AdditionModel(nn.Module):
    def forward(self, x):
        return x + 1

addition_model = AdditionModel()
addition_mh = tl.log_forward_pass(addition_model, x)
print(addition_mh.output_layers) # ['output_1']

addition_mh.save_new_activations(addition_model, x)
print(addition_mh.output_layers) # ['output_1', 'add_1_2_raw']
```

**Comment by johnmarktaylor91** [comment]
Thanks for finding this—I'll take a look pronto.

**Comment by raktes** [comment]
Thanks. I also noticed that the tensor_content of the output TensorLogEntry is None after save_new_activations, which is probably related to the issue.

**Comment by johnmarktaylor91** [comment]
Okay, should now be fixed on both fronts (the extra layers in `output_layers`, and the tensor contents being `None`).

**Comment by raktes** [comment]
Yes, both problems seem to be fixed now.

However, a small bug remains. For the BasicModel in the code example above, the number of saved tensors, and therefore also the space of saved tensors seems to be wrong, as well as the check if all layers are saved. Printing the ModelHistory yields:

```
	Tensor info:
		- 2 total tensors (304 B) computed in forward pass.
		- 2 tensors (304 B) with saved activations.
	Parameters: 0 parameter operations (0 params total; 0 B)
	Module Hierarchy:
	Layers (all have saved activations):
		  (0) input_1 
		  (1) output_1 
```

while the ModelHistory after save_new_activations contained
```
	Tensor info:
		- 2 total tensors (304 B) computed in forward pass.
		- 1 tensors (152 B) with saved activations.
	Parameters: 0 parameter operations (0 params total; 0 B)
	Module Hierarchy:
	Layers (* means layer has saved activations):
		* (0) input_1 
		* (1) output_1 
```

The same bug doesn't appear for the AdditionModel from the code example above. Not sure if this has to do with the problems regarding the output layer logic, or if it is a separate issue.

**Comment by johnmarktaylor91** [comment]
Good catch, fixed this too. Thanks for your careful help with debugging this! I should mention that I'm planning a refactor of `save_new_activations` soon-ish, which should hopefully help clean up any remaining little bugs like this. The reason for this whack-a-mole is that there are a lot of fields logged by TorchLens, some of which are updated after save_new_activations and some of which stay the same, and there were some cases where they weren't being updated in a consistent way (which you've helpfully found). Much appreciate you bringing all this to my attention.

**Comment by raktes** [comment]
No Problem, love this library so far, it's really useful. The issue is fixed now, thanks.
